cmake/config.cmake

```
@@ -146,8 +144,20 @@ set(OMR_GC_SCAN_OBJECT_GLUE OFF CACHE BOOL "Implement ScanObject in glue code, n
 set(OMR_GC_SEGREGATED_HEAP OFF CACHE BOOL "TODO: Document")
 set(OMR_GC_VLHGC OFF CACHE BOOL "TODO: Document")
 set(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD OFF CACHE BOOL "Enable VLHGC concurrent copy forward")
-
 set(OMR_INTERP_HAS_SEMAPHORES ON CACHE BOOL "TODO: Document")
+set(OMR_GC_POINTERS "full" CACHE BOOL "The mode of pointers.  Must be \"full\", \"compressed\", or \"mixed\"")
+if(OMR_GC_POINTERS STREQUAL "full")
+	set(OMR_GC_COMPRESSED_POINTERS OFF CACHE BOOL "")
+	set(OMR_GC_FULL_POINTERS ON CACHE BOOL "")
+elseif (OMR_GC_POINTERS STREQUAL "compressed")
+	set(OMR_GC_COMPRESSED_POINTERS ON CACHE BOOL "")
+	set(OMR_GC_FULL_POINTERS OFF CACHE BOOL "")
+elseif (OMR_GC_POINTERS STREQUAL "mixed")
+	set(OMR_GC_COMPRESSED_POINTERS ON CACHE BOOL "")
+	set(OMR_GC_FULL_POINTERS ON CACHE BOOL "")
+else()
+	omr_assert(FATAL_ERROR MESSAGE "OMR_GC_FULL_POINTERS must be set to one of \"full\", \"compressed\", or \"mixed\"")
```

since there's no condition, this can be a plain ol' message:
```suggestion
	message(FATAL_ERROR "OMR_GC_FULL_POINTERS must be set to one of \"full\", \"compressed\", or \"mixed\"")
```